pysteps/feature/tstorm.py

```diff
@@ -0,0 +1,320 @@
+    import skimage.morphology as skim
+    import skimage.segmentation as skis
+try:
+    import pandas as pd
+
+    PANDAS_IMPORTED = True
+except ImportError:
+    PANDAS_IMPORTED = False
+
+
+def detection(
+    input_image,
+    minref=35,
+    maxref=48,
+    mindiff=6,
+    minsize=50,
+    minmax=41,
+    mindis=10,
+    dyn_thresh=False,
+    output_feat=False,
+    time="000000000",
+):
+    """
+    This function detects thunderstorms using a multi-threshold approach. It is
+    recommended to use a 2-D Cartesian maximum reflectivity composite, however the
+    function will process any 2-D array.
+    The thunderstorm cell detection requires both scikit-image and pandas.
+
+    Parameters
+    ----------
+    input_image : array-like
+        Array of shape (m,n) containing input image, usually maximum reflectivity in
+        dBZ with a resolution of 1 km. Nan values are ignored.
+    minref : float, optional
+        Lower threshold for object detection. Lower values will be set to NaN.
+        The default is 35 dBZ.
+    maxref : float, optional
+        Upper threshold for object detection. Higher values will be set to this value.
+        The default is 48 dBZ.
+    mindiff : float, optional
+        Minimal difference between two identified maxima within same area to split area
+        into two objects. The default is 6 dBZ.
+    minsize : float, optional
+        Minimal area for possible detected object. The default is 50 pixels.
+    minmax : float, optional
+        Minimum value of maximum in identified objects. Objects with a maximum lower
+        than this will be discarded. The default is 41 dBZ.
+    mindis : float, optional
+        Minimum distance between two maxima of identified objects. Objects with a
+        smaller distance will be merged. The default is 10 km.
+    dyn_thresh: bool, optional
+        Set to True to activate dynamic lower threshold. Restricts contours to more
+        meaningful area. The default is False.
+    output_feat: bool, optional
+        Set to True to return only the cell coordinates.
+    time : string, optional
+        Date and time as string. Used to label time in the resulting dataframe.
+        The default is '000000000'.
+
+    Returns
+    -------
+    cells_id : pandas dataframe
+        Pandas dataframe containing all detected cells and their respective properties
+        corresponding to the input image.
+    labels : array-like
+        Array of shape (m,n), grid of labelled cells.
+    """
+    if not SKIMAGE_IMPORTED:
+        raise MissingOptionalDependency(
+            "skimage is required for thunderstorm DATing " "but it is not installed"
+        )
+    if not PANDAS_IMPORTED:
+        raise MissingOptionalDependency(
+            "pandas is required for thunderstorm DATing " "but it is not installed"
+        )
+    filt_image = np.zeros(input_image.shape)
+    filt_image[input_image > minref] = input_image[input_image > minref]
```

Comment:
According to the definition of minder the help, the comparison operator should be equal or greater than minref. Correct?

Reply:
I corrected it to >= in the code in line 120.